### PR TITLE
refactor/bd 7 add input component

### PR DESCRIPTION
**Pull request by SlavaMryasov**

### CODE REVIEW

File: src/components/input/Input.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+} satisfies Meta<typeof Input>
+
+export default meta
+
+export const Default: StoryObj<typeof meta> = {
+  args: {
+    disabled: true,
+    errorText: '',
+    // iconStart: (
+    //   <Icon cursor='pointer' height={20} icon='search-outline' width={20} />
```
Comment (sfgardev) [inline src/components/input/Input.stories.tsx:19]: remove commented code

File: src/components/input/Input.tsx [hunk 2]
```diff
@@ -1,82 +1,83 @@
...
-          {type === 'password' && currentType === 'text' && (
-            <Icon className={iconColor} icon='eye-outline' />
-          )}
-          {type === 'search' && (
-            <Icon className={iconColor} icon='search-outline' />
+          <input
+            disabled={disabled}
+            id={id ?? myId}
+            ref={ref}
+            {...props}
```
Comment (sfgardev) [inline src/components/input/Input.tsx:47]: move {...props} to the end (after all another props)

File: src/components/input/PasswordInput.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+export const PasswordInput = forwardRef<HTMLInputElement, InputProps>(
+  (
+    { disabled, label = 'password', placeholder = 'enter password', ...props },
+    ref
+  ) => {
+    const [visible, setVisible] = useState(true)
+
+    return (
+      <Input
+        {...props}
```
Comment (sfgardev) [inline src/components/input/PasswordInput.tsx:16]: move props to the end

File: src/components/input/types/inputTypes.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { ComponentProps, ReactNode } from 'react'
```
Comment (sfgardev) [inline src/components/input/types/inputTypes.ts:1]: remove inputtypes from folder types and rename it to types.ts

File: src/components/input/Input.tsx [hunk 5]
```diff
@@ -1,82 +1,83 @@
...
-          disabled={disabled}
-          id={id ?? myId}
-          placeholder={placeholder}
-          type={currentType}
-          value={props.value}
-        />
+    return (
+      <div className='flex flex-col'>
+        {label && (
+          <label
```
Comment (sfgardev) [inline src/components/input/Input.tsx:17]: add htmlFor attribute for connecting input and label

File: src/components/input/PasswordInput.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import { forwardRef, useState } from 'react'
+
+import { Icon } from '../icon'
+import { Input } from './Input'
+import { InputProps } from './types/inputTypes'
+
+export const PasswordInput = forwardRef<HTMLInputElement, InputProps>(
+  (
+    { disabled, label = 'password', placeholder = 'enter password', ...props },
```
Comment (sfgardev) [inline src/components/input/PasswordInput.tsx:9]: Label and placeholder start with uppercase letter

File: src/components/input/PasswordInput.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    ref
+  ) => {
+    const [visible, setVisible] = useState(true)
```
Comment (sfgardev) [inline src/components/input/PasswordInput.tsx:12]: add more specificities for naming state (isPasswordVisible for exmaple)

**Review by sfgardev — CHANGES_REQUESTED** [review]

File: src/components/input/Input.stories.tsx (hunk 1 above)
Comment (SlavaMryasov) [reply src/components/input/Input.stories.tsx:19]: this is for example

File: src/components/input/Input.tsx (hunk 2 above)
Comment (SlavaMryasov) [reply src/components/input/Input.tsx:47]: done

File: src/components/input/PasswordInput.tsx (hunk 3 above)
Comment (SlavaMryasov) [reply src/components/input/PasswordInput.tsx:16]: done

File: src/components/input/types/inputTypes.ts (hunk 4 above)
Comment (SlavaMryasov) [reply src/components/input/types/inputTypes.ts:1]: done

File: src/components/input/Input.tsx (hunk 5 above)
Comment (SlavaMryasov) [reply src/components/input/Input.tsx:17]: sorry, i'm fogot. done!

File: src/components/input/PasswordInput.tsx (hunk 6 above)
Comment (SlavaMryasov) [reply src/components/input/PasswordInput.tsx:9]: done

File: src/components/input/PasswordInput.tsx (hunk 7 above)
Comment (SlavaMryasov) [reply src/components/input/PasswordInput.tsx:12]: done

**Review by sfgardev — APPROVED** [review]

**Pull request merged**